prisma/schema.prisma

```diff
@@ -115,16 +119,100 @@ model Area {
   stageId              Int
-  stage                Stage      @relation(fields: [stageId], references: [id])
+  stage                Stage            @relation(fields: [stageId], references: [id])
   artefacts            Artefact[]
+  AssessmentArea       AssessmentArea[]
+  Score                Score[]
 
   @@unique([area_id, area_name, stageId])
 }
 
-
 model Stage {
   stageNumber Int
-  id          Int    @id @default(autoincrement())
-  name        String @unique
+  id          Int     @id @default(autoincrement())
+  name        String  @unique
   areas       Area[]
+  Score       Score[]
+}
+
+// New Models
+
+model Project {
+  id          Int           @id @default(autoincrement())
+  projectId   String        @unique
+  name        String
+  createdAt   DateTime      @default(now())
+  updatedAt   DateTime      @updatedAt
+  assessments Assessment[]
+  users       UserProject[]
+}
+
+model UserProject {
+  userId    String
+  projectId Int
+  user      User    @relation(fields: [userId], references: [id], onDelete: Cascade)
+  project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+  @@id([userId, projectId])
+}
+
+model Assessment {
+  id        Int                  @id @default(autoincrement())
+  projectId Int
+  userId    String
+  createdAt DateTime             @default(now())
+  updatedAt DateTime             @updatedAt
+  project   Project              @relation(fields: [projectId], references: [id], onDelete: Cascade)
+  user      User                 @relation(fields: [userId], references: [id])
+  artefacts AssessmentArtefact[]
+  areas     AssessmentArea[]
+  Score     Score[]
+}
+
+model AssessmentArtefact {
+  id           Int        @id @default(autoincrement())
+  artefactId   Int
+  assessmentId Int
+  checkmark    Boolean
+  comment      String?
+  artefact     Artefact   @relation(fields: [artefactId], references: [id])
+  assessment   Assessment @relation(fields: [assessmentId], references: [id], onDelete: Cascade)
+
+  @@unique([artefactId, assessmentId])
+}
+
+model AssessmentArea {
+  id           Int                      @id @default(autoincrement())
+  areaId       Int
+  assessmentId Int
+  comment      String?
+  score        Float
+  area         Area                     @relation(fields: [areaId], references: [id])
+  assessment   Assessment               @relation(fields: [assessmentId], references: [id], onDelete: Cascade)
+  questions    AssessmentAreaQuestion[]
+
+  @@unique([areaId, assessmentId])
+}
+
+model AssessmentAreaQuestion {
+  id               Int            @id @default(autoincrement())
+  assessmentAreaId Int
+  question         String
+  answer           Int?
+  assessmentArea   AssessmentArea @relation(fields: [assessmentAreaId], references: [id], onDelete: Cascade)
+}
+
+model Score {
+  id           Int        @id @default(autoincrement())
+  assessmentId Int
+  areaId       Int?
+  artefactId   Int?
+  stageId      Int?
+  score        Float
```

Comment:
We for sure should differ between artefact maturity score and area maturity score. It comes as an additional field only in `assessment + area`, `assessment + stage`, but not in `assessment + artefact`

I see this as a downside of this approach. At first thought, maybe having ArtefactBasedScore and AreaBasedScore would solve the problem

Reply:
This part here I keep questioning myself in. If you have a clearer image about it, let's just follow your implementation @severynenko